_includes/v19.1/app/insecure/gorm-sample.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"math"
+	"math/rand"
+	"time"
+
+	// Import GORM-related packages.
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/postgres"
+
+	// Necessary in order to check for transaction retry error codes.
+	// See implementation below in `transferFunds`.
+	"github.com/lib/pq"
+)
+
+// Account is our model, which corresponds to the "accounts" database
+// table.
+type Account struct {
+	ID      int `gorm:"primary_key"`
+	Balance int
+}
+
+func main() {
+	// Connect to the "bank" database as the "maxroach" user.
+	const addr = "postgresql://maxroach@localhost:26257/bank?sslmode=disable"
+	db, err := gorm.Open("postgres", addr)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+
+	// Set to `true` and GORM will print out all DB queries.
+	db.LogMode(false)
+
+	// Automatically create the "accounts" table based on the Account
+	// model.
+	db.AutoMigrate(&Account{})
+
+	// Insert two rows into the "accounts" table.
+	db.Create(&Account{ID: 1, Balance: 1000})
+	db.Create(&Account{ID: 2, Balance: 250})
+
+	// The sequence of steps in this section is:
+	// 1. Print account balances.
+	// 2. Set up some Accounts and transfer funds between them.
+	// 3. Print account balances again to verify the transfer occurred.
+
+	// Print balances before transfer.
+	printBalances(db)
+
+	var amount = 100
+	var fromAccount Account
+	var toAccount Account
+
+	db.First(&fromAccount, 1)
```

Comment:
These calls must be inside the transaction. Any time the transaction retries, we need to re-query the starting balances. (This applies to the other language examples too).

Reply:
Fixed these by moving them inside `transferFunds`, which is now a function argument to a `runTransaction` function very similar to the Python examples.